sources/api/migration/migration-helpers/src/common_migrations.rs

```diff
@@ -4,33 +4,58 @@ use serde::Serialize;
 use snafu::{OptionExt, ResultExt};
 use std::collections::HashMap;
 
-/// We use this migration when we add a setting and want to make sure it's removed before we go
-/// back to old versions that don't understand it.
-pub struct AddSettingMigration(pub &'static str);
+/// We use this migration when we add settings and want to make sure they're removed before we go
+/// back to old versions that don't understand them.
+pub struct AddSettingsMigration<'a> (pub &'a [&'static str]);
 
-impl Migration for AddSettingMigration {
-    /// New versions must either have a default for the setting or generate it; we don't need to
+impl Migration for AddSettingsMigration<'_> {
+    /// New versions must either have a default for the settings or generate them; we don't need to
     /// do anything.
     fn forward(&mut self, input: MigrationData) -> Result<MigrationData> {
-        println!("AddSettingMigration({}) has no work to do on upgrade.", self.0);
+        println!(
+            "AddSettingsMigration({:?}) has no work to do on upgrade.",
+            self.0
+        );
         Ok(input)
     }
 
-    /// Older versions don't know about the setting; we remove it so that old versions don't see
-    /// it and fail deserialization.  (The setting must be defaulted or generated in new versions,
+    /// Older versions don't know about the settings; we remove them so that old versions don't see
+    /// them and fail deserialization.  (The settings must be defaulted or generated in new versions,
     /// and safe to remove.)
     fn backward(&mut self, mut input: MigrationData) -> Result<MigrationData> {
-        if let Some(data) = input.data.remove(self.0) {
-            println!("Removed {}, which was set to '{}'", self.0, data);
-        } else {
-            println!("Found no {} to remove", self.0);
+        for setting in self.0 {
+            if let Some(data) = input.data.remove(*setting) {
+                println!("Removed {}, which was set to '{}'", setting, data);
+            } else {
+                println!("Found no {} to remove", setting);
+            }
         }
         Ok(input)
     }
 }
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// Similar to the above, this migration is for when we add a single setting.
+/// We are retaining this migration helper in case there are migrations already using it.
+pub struct AddSettingMigration(pub &'static str);
+
+impl Migration for AddSettingMigration {
+    fn forward(&mut self, input: MigrationData) -> Result<MigrationData> {
+        println!(
+            "AddSettingMigration({}) has no work to do on upgrade.",
+            self.0
+        );
+        Ok(input)
```

Comment:
```suggestion
        AddSettingsMigration(&[self.0]).forward(input)
```
I'd suggest we simplify by instead wrapping the existing functionality (even though it prints the other struct name) and add a `#[deprecated(...)]` marker to this struct to make it clear that it's been replaced.